app/controllers/support_interface/providers_controller.rb

```diff
@@ -14,7 +14,11 @@ def show
     end
 
     def courses
-      @provider = Provider.includes(:courses).find(params[:provider_id])
+      @provider = Provider.includes(courses: [:accrediting_provider]).find(params[:provider_id])
+
+      accredited_providers = @provider.courses.map(&:accrediting_provider).uniq.compact
+
+      @all_accredited_providers_onboarded = accredited_providers.all?(&:onboarded?)
```

Comment:
Perhaps these two lines can be extracted into a service/Provider model so you can add a unit test?

Reply:
Sure — maybe `Provider` isn't a bad place for this to live